openedx/core/djangoapps/oauth_dispatch/tests/test_views.py

```diff
@@ -174,16 +174,22 @@ def _test_jwt_access_token(self, client_attr, token_type=None, headers=None, gra
         expected_default_expires_in = 60 * 60
         assert data['expires_in'] == expected_default_expires_in
         assert data['token_type'] == 'JWT'
+        expected_scopes = data['scope'].split(' ')
+        self._update_expected_scopes_with_user_id(expected_scopes, grant_type)
```

Comment:
I need help understanding how this isn't just covering up a problem. Shouldn't `data` which comes from the response contain all the necessary scopes? Something feels fishy about this fix, so maybe you can help explain it a bit better. Thanks.

Reply:
I think this is a bug:
* The scopes in the response used to match the scopes in the JWT, but now they don't. 
* I _think_ the fix would be that [this comment should be removed](https://github.com/openedx/edx-platform/blob/5ce6c4d11956fe59dff67243068ef4540fa4e6ab/openedx/core/djangoapps/oauth_dispatch/jwt.py#L95), and that we now need to set the scopes in the response to the scopes in the JWT.

Let me know your thoughts.

Reply:
If you call the API from the command line using a password grant type, is the user_id in the scopes in the dict of the response? This dict should be in sync with the scopes in the JWT, and I’m guessing it’s not, but I could be wrong.

Reply:
I am hoping that none of this new test code would be needed, because we are testing that the API response scopes and JWT scopes match, and both should have the new user_id, right?